Common/util/directory.h

```diff
@@ -35,11 +28,11 @@ namespace Common
 namespace Directory
 {
     // Creates new directory (if it does not exist)
-    bool   CreateDirectory(const String &path);
+    bool   EnsureDirectoryExists(const String &path);
```

Comment:
Is this really necessary? We do abide the convention by placing our functions in our own namespace. I'd hate to force inventing different names just because someone else did not abide this convention and declared macros of commonly used names.

Additionally, I believed it may be possible to reduce places where windows headers are included to reduce (if not completely prevent) the conflicts.